### DISCUSSION TITLE
Add a custom fail_timeout value

### DISCUSSION THREAD

**Discussion by oom-** [nginx-proxy/nginx-proxy#2403] [Q&A]
Hello, 

Is there any way to add a fail_timeout value in the upstream section ? 
Doesn't seems to be "editable" ?

Thanks.

**Comment by SchoNie** [comment]
Not in an easy configurable method currently.
If you need it for all your servers, a quick and dirty workaround could be to mount a custom `nginx.tmpl` and hardcode the fail_timeout parameter [at line 315](https://github.com/nginx-proxy/nginx-proxy/blob/main/nginx.tmpl#L315)

We could make a PR for this as I see more use cases, for example define this `fail_timeout` value but also a `weight` if combined with the loadbalance feature.
For example make a `com.github.nginx-proxy.nginx-proxy.upstream-parameters` container label for advanced use cases.
Not sure if this is something @buchdag would consider merging? (If interested, I can make a PR)

**Reply by buchdag** [reply]
@SchoNie as long as it is tested, documented, and not a marginal use case (I don't think it is if you had in mind a generalised way to pass parameters to the `server` directive), I'll consider it. But the template is probably going to be heavily refactored as we work toward #2130 / #1504, so I can't guarantee that working on this right now won't end up with merge conflicts in the PR ☹️
